src/megaapi_impl.cpp

```diff
@@ -22752,6 +22829,89 @@ bool MegaApiImpl::tryLockMutexFor(long long time)
     }
 }
 
+void MegaApiImpl::setBackup(int backupType, MegaHandle targetNode, const char* localFolder, const char* deviceId, int state, int subState, const char* extraData, MegaRequestListener* listener)
+{
+    MegaRequestPrivate *request = new MegaRequestPrivate(MegaRequest::TYPE_BACKUP_PUT, listener);
+    request->setTotalBytes(backupType);
+    request->setNodeHandle(targetNode);
+    request->setFile(localFolder);
+    request->setName(deviceId);
+    request->setAccess(state);
+    request->setNumDetails(subState);
+    request->setText(extraData);
+    request->setListener(listener);
+    requestQueue.push(request);
+    waiter->notify();
+}
+
+void MegaApiImpl::removeBackup(MegaHandle backupId, MegaRequestListener *listener)
+{
+    MegaRequestPrivate* request = new MegaRequestPrivate(MegaRequest::TYPE_BACKUP_REMOVE, listener);
+    request->setParentHandle(backupId);
+    request->setListener(listener);
+    requestQueue.push(request);
+    waiter->notify();
+}
+
+void MegaApiImpl::updateBackup(MegaHandle backupId, int backupType, MegaHandle targetNode, const char* localFolder, const char* deviceId, int state, int subState, const char* extraData, MegaRequestListener* listener)
+{
+    MegaRequestPrivate* request = new MegaRequestPrivate(MegaRequest::TYPE_BACKUP_PUT, listener);
+    request->setParentHandle(backupId);
+
+    if (backupType != BackupType::INVALID)
+    {
+        request->setTotalBytes(backupType);
+    }
+
+    if (targetNode != UNDEF)
+    {
+        request->setNodeHandle(targetNode);
+    }
+
+    if (localFolder)
+    {
+        request->setFile(localFolder);
+    }
+
+    if (deviceId)
+    {
+        request->setName(deviceId);
+    }
+
+    if (state > 0)
+    {
+        request->setAccess(state);
+    }
+
+    if (subState > 0)
+    {
+        request->setNumDetails(subState);
+    }
```

Comment:
Those should be allowed to be 0. 
See https://github.com/meganz/sdk/pull/2112#pullrequestreview-483861021